File: Ruby/hh-credit-autoload-ruby/helpers/hello-hunter-api.rb

class HelloHunterApi
  attr_accessor :conn, :token, :response, :customers

  def initialize
    self.conn = Faraday.new "https://SQL-04.dialer.rocks/a/v1/resellers/" do |builder|
      builder.request :json
      builder.response :json
      builder.response :raise_error
      builder.response :logger
      builder.headers['Content-Type'] = 'application/json'
    end
    self.connect
    self
  end

  def connect
    creds = {name: ENV['HH_USERNAME'], pass: ENV['HH_PASSWORD']}
    begin
      response = @conn.post('login.json', creds.to_json)
      response_body = response.body
      @token = response_body['token']
      @customers = response_body['user_groups']
      $logger.info('Hello Hunter login successful')
    rescue Faraday::Error => e
      $logger.error('Hello Hunter Token not received')
      $logger.debug e.response[:status]
      $logger.debug e.response[:body]
    end
  end

  def give_credit(z_hh_id, reload_amount)
    params = {user_group_id: z_hh_id, to_add: reload_amount}
    begin
      response = @conn.put("add_balance.json?token=#{@token}", params.to_json)
      $logger.info("Reload api call success. Amount added #{response['amount']}")
      return response.status
    rescue Faraday::Error => e
      $logger.error("Hello Hunter reload of $#{reload_amount} failed for #{z_hh_id}")
      $logger.debug e.response[:status]
      $logger.debug e.response[:body]
      return nil
    end
  end
end

File: hh-credit-autoload-ruby/functions/hello-hunter-api.js
const axios = require('axios');
require('dotenv').config();

class HelloHunterApi {
    constructor() {
        this.baseUrl = 'https://SQL-04.dialer.rocks/a/v1/resellers/';
        this.conn = axios.create({
            baseURL: this.baseUrl,
            headers: {
                'Content-Type': 'application/json',
            },
        });
        this.token = null;
        this.customers = null;
    }

    async connect() {
        const creds = {
            name: process.env.HH_USERNAME,
            pass: process.env.HH_PASSWORD,
        };

        try {
            const response = await this.conn.post('login.json', creds);
            const { token, user_groups: customers } = response.data;
            this.token = token;
            this.customers = customers;
            console.info('Hello Hunter login successful');
        } catch (error) {
            console.error('Hello Hunter Token not received');
            if (error.response) {
                console.debug(error.response.status);
                console.debug(error.response.data);
            } else {
                console.error(error.message);
            }
        }
    }

    async giveCredit(z_hh_id, reloadAmount) {
        const params = {
            user_group_id: z_hh_id,
            to_add: reloadAmount,
        };

        try {
            const response = await this.conn.put(
                `add_balance.json?token=${this.token}`,
                params
            );
            console.info(`Reload API call success. Amount added: ${response.data.amount}`);
            return response.status;
        } catch (error) {
            console.error(`Hello Hunter reload of $${reloadAmount} failed for ${z_hh_id}`);
            if (error.response) {
                console.debug(error.response.status);
                console.debug(error.response.data);
            } else {
                console.error(error.message);
            }
            return null;
        }
    }
}

module.exports = HelloHunterApi;
